pkg/kubectl/cmd/util/shortcut_restmapper.go

```diff
@@ -124,17 +133,33 @@ func (e ShortcutExpander) AliasesForResource(resource string) ([]string, bool) {
 		}
 		return aliases, true
 	}
-	expanded := expandResourceShortcut(schema.GroupVersionResource{Resource: resource}).Resource
+	expanded := e.expandResourceShortcut(schema.GroupVersionResource{Resource: resource}).Resource
 	return []string{expanded}, (expanded != resource)
 }
 
-// expandResourceShortcut will return the expanded version of resource
+// expandResourceShortcut will return the expanded version of a resource
 // (something that a pkg/api/meta.RESTMapper can understand), if it is
-// indeed a shortcut. Otherwise, will return resource unmodified.
-func expandResourceShortcut(resource schema.GroupVersionResource) schema.GroupVersionResource {
-	if expanded, ok := kubectl.ShortForms[resource.Resource]; ok {
-		resource.Resource = expanded
-		return resource
+// indeed a shortcut. First the list of potential resources will be taken from
+// the discovery API. Next we will fall back to the hardcoded list of resources.
+// Lastly if no match has been found, we will return resource unmodified.
+func (e ShortcutExpander) expandResourceShortcut(gvk schema.GroupVersionResource) schema.GroupVersionResource {
+	// get the server resources if no gv was specified examine all and return
+	// on first match.
+	if resources, err := e.getServerRes(gvk.GroupVersion().String()); err == nil {
+		for _, resource := range resources {
+			for _, apiRes := range resource.APIResources {
+				if apiRes.ShortName == gvk.Resource {
+					gvk.Resource = apiRes.Name
+					return gvk
+				}
+			}
+		}
+	}
+
+	// fall back to the hardcoded list
+	if expanded, ok := kubectl.ShortForms[gvk.Resource]; ok {
```

Comment:
I think the shortforms need to be updated to have the same logical ordering and tupled with group.